tools/test.py

```diff
@@ -1273,6 +1276,24 @@ def ProcessOptions(options):
   VERBOSE = options.verbose
   options.arch = options.arch.split(',')
   options.mode = options.mode.split(',')
+  options.run = options.run.split(',')
+  if options.run == [""]:
+    options.run = None
+  elif len(options.run) != 2:
+    print "The run argument must be two comma-separated integers."
+    return False
+  else:
+    try:
+      options.run = map(int, options.run)
+    except ValueError:
+      print "Could not parse the integers from the run argument."
+      return False
+    if options.run[0] < 0 or options.run[1] < 0:
+      print "The run argument cannot have negative integers."
+      return False
+    if options.run[0] >= options.run[1]:
+      print "The test to run (n) must be smaller than the modulo (m)."
```

Comment:
The goal here is to divide the test to run into groups and run only a specific test from each group. Having n >= m could indeed run as you stated above, but there is no practical application for it. If you specify n >= m, it's a bug and this should fail with an error.

If the starting test is greater than the number of tests to be executed, it will just say `No tests to run.`. That is coherent with what happens in other cases, like specifying an empty test suite.